api/src/main/proto/stellarstation/api/v1/groundstation/groundstation.proto

```diff
@@ -159,6 +159,23 @@ message UnavailabilityWindow {
 
   // End time of the unavailability window.
   google.protobuf.Timestamp end_time = 3;
+
+  // Indicates whether this window is part of a recurring set.
+  enum RecurringPattern {
+    // This is a one-off window that does not recur.
+    NONE = 0;
+
+    // The window recurs daily at the same time.
+    DAILY = 3;
+
+    // The window recurs weekly on the same day.
+    WEEKLY = 1;
+  }
+
+  // Details of the recurrence pattern.
+  //
+  // Note that 'window_id' is the same for all instances of a recurring window.
```

Comment:
I just want to clarify the behavior. Will the API return instances of recurring windows?
If so, we will need a max_instance field with default value.

Reply:
Originally I was going to have one UnavailabilityWindow message per 'instance'. However, Dan pointed out that this causes a change in behavior for scripts that might try to delete windows based on collecting the window IDs (all instances in the recurring set will have the same ID). The alternative is to have some repeated field inside the UnavailabilityWindow message itself.

I'm not sure I understand what you mean by max_instance though. Recurring windows are infinite but this API is restricted by the time range in the request.

Reply:
We might be able to add a repeated message in the future named instance like Google Calendar API but for now, I think it's good to just return one UnavailabilityWindow. 
https://developers.google.com/calendar/recurringevents

About max_instance, never mind about it. I totally missed the fact that the request message has start_time and end_time.

Reply:
Having expandable windows have a nested message with the recurring seems like a clean API to me. Such structure is more difficult to express in the CLI, but the API's design is the #1 priority.